addons/sale/models/sale.py

```diff
@@ -1635,8 +1642,15 @@ def product_uom_change(self):
                 uom=self.product_uom.id,
                 fiscal_position=self.env.context.get('fiscal_position')
             )
-            company = self.company_id or self.order_id.company_id
-            self.price_unit = self.env['account.tax']._fix_tax_included_price_company(self._get_display_price(product), product.taxes_id, self.tax_id, company)
+            self.price_unit = product._get_default_price_unit(
+                self.company_id or self.order_id.company_id,
```

Comment:
I'm not a sale expert but it seems really weird... both should be always the same, right?

Reply:
I think it could happen for the related field to not have the correct order value when editing/creating line in the interface.
At least in older versions, but I cannot say for sure it's still needed, didn't see the problem since quite some time now (I suppose it's fixed in master)

Anyway, doesn't break/cost anything, it was the previous behavior, let's keep it like that, changes should be as limited as possible in stable.

Reply:
I'll second this, when creating the line sometimes the `company_id` is not set and we refer to the sale order company (somewhere even the env company if also the sale order is still a virtual record)